### DISCUSSION TITLE
[Docs] SSH Key Configuration for WSL 2 Setup

### DISCUSSION THREAD

**Discussion by Maarten-Dekker** [serversideup/spin#94] [Ideas]
# 👉 The problem
Setting up a new WSL 2 environment and proceeding with the `spin version` command works flawlessly. However, the next step, which involves initializing a project with `spin init <template>` or `spin new laravel my-billion-dollar-idea`, hits a snag due to the absence of an SSH key linked to the GitHub account. This results in the following error:

```
🔄 Downloading template...
Cloning from git@github.com:serversideup/spin-template-laravel-basic.git
Cloning into '/tmp/tmp.bFMHoirBnG'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

This issue prevents any further progress in the installation process.

# 🏆 How to solve this problem
To address this, include a prerequisite step in the documentation. Users should be guided to configure an SSH key in their GitHub account before initiating a project setup. The following steps can be added to the documentation to facilitate this process:

1. Verify if an SSH key is already present: [Check for existing SSH keys](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys)
2. Generate a new SSH key: [Generate a new SSH key and add it to the SSH agent](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
3. Add the new SSH key to your GitHub account: [Add a new SSH key to your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
4. Test your SSH connection: [Test your SSH connection](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/testing-your-ssh-connection)

Incorporating these steps ensures users have the necessary access rights, thereby avoiding the aforementioned error and smoothing out the project initialization phase.

Note: i am not well-versed in WSL or Linux, there could potentially be a simpler method. However, the steps I've outlined have effectively resolved the issue for me.

**Comment by jaydrogers** [comment]
This was fixed in v2.1.0. Thanks for submitting this! https://github.com/serversideup/spin/releases/tag/v2.1.0
